webdavd.c

```diff
@@ -1912,6 +1926,61 @@ static void initializeEnvVariables() {
 	else unsetenv("WEBDAVD_CHROOT_PATH");
 }
 
+static void freePassedSockets(PassedSockets * ps) {
+	if (ps) {
+		for (int i = 0; i < ps->count; i++)
+			free(ps->socket[i].name); // Allocated by libsystemd
+		freeSafe(ps); // Allocated by us
+	}
+}
+
+#if HAVE_SYSTEMD
+static void initializeSocketActivation(PassedSockets ** socketsRet) {
+	if (!config.socketActivation) {
+		*socketsRet = NULL;
+		return;
+	}
+
+	char **names;
+	// This will set close on exec on all sockets
+	int n = sd_listen_fds_with_names(1, &names);
+	if (n <= 0) {
+		*socketsRet = NULL;
+		return;
+	}
+	PassedSockets * ps = mallocSafe(sizeof(*ps) + sizeof(*ps->socket) * n);
+	ps->count = n;
+
+	bool error = false;
+	for (int i = 0; i < n; i++) {
+		ps->socket[i].fd = SD_LISTEN_FDS_START + i;
+		ps->socket[i].name = names[i]; // Take ownership of string, need to free on exit
+		ps->socket[i].used = false;
+		int ret = sd_is_socket(ps->socket[i].fd, AF_UNSPEC, SOCK_STREAM, true);
+		if (ret <= 0) {
+			stdLogError(-ret, "Socket #%d '%s' unstuitable for WebDAV", i, names[i]);
+			error = true;
+		}
+	}
+	free(names);
+
+	if (error) {
+		// Just clean up stuct now and refuse to use any sockets
+		freePassedSockets(ps);
+		ps = NULL;
+	}
```

Comment:
I have this totally fail if any socket passed is not acceptable.  I'd rather have something fail if there is any problem than to keep going and maybe work partially.

But your design elsewhere doesn't seem to be like this.  I.e., failure to start a daemon will continue to the next one and not return failure. 

This code here could easily change to be like that too.  The code related to `error` can just be deleted.  Any other sockets will still work fine.  Maybe the config xml doesn't even use the bad socket for any listen sections and it doesn't matter?  If MHD doesn't like the socket, it'll give us an error and that listener won't start.